### ISSUE TITLE
Error: cec-client never reported ready

### ISSUE THREAD

**Issue by sindudas** [jbree/homebridge-plugin-cec#4]
I got it working one time, but not powering on or powering off the TV. But now the log is dumping that messages:

Jan 24 23:06:55 homebridge[2438]: Error: cec-client never reported ready
Jan 24 23:06:55 homebridge[2438]: at Timeout.<anonymous> (/usr/local/lib/node_modules/homebridge-platform-cec/node_modules/cec-promise/index.js:13:12)
Jan 24 23:06:55 homebridge[2438]: at ontimeout (timers.js:365:14)
Jan 24 23:06:55 homebridge[2438]: at tryOnTimeout (timers.js:237:5)
Jan 24 23:06:55 homebridge[2438]: at Timer.listOnTimeout (timers.js:207:5)
Jan 24 23:06:55 homebridge[2438]: Error: cec-client never reported ready
Jan 24 23:06:55 homebridge[2438]: at Timeout.<anonymous> (/usr/local/lib/node_modules/homebridge-platform-cec/node_modules/cec-promise/index.js:13:12)
Jan 24 23:06:55 homebridge[2438]: at ontimeout (timers.js:365:14)
Jan 24 23:06:55 homebridge[2438]: at tryOnTimeout (timers.js:237:5)
Jan 24 23:06:55 homebridge[2438]: at Timer.listOnTimeout (timers.js:207:5)

And now can't switch on or off on my iPhone.

Reboot the rpi completly several times, but nothing, always the same.

**Comment by jbree** [comment]
Hmm this might be the same thing as #1 . Would you mind providing specifics about OS version on your rpi, and the version of cec-client you have installed also?

I've run into this problem before during development when I was constantly restarting homebridge, but a restart always fixed it.

**Comment by sindudas** [comment]
It's a RPi 1

> $ cat /etc/debian_version 
 8.0

> $ npm version
{ npm: '3.10.10',
  ares: '1.10.1-DEV',
  http_parser: '2.7.0',
  icu: '57.1',
  modules: '48',
  node: '6.9.4',
  openssl: '1.0.2j',
  uv: '1.9.1',
  v8: '5.1.281.89',
  zlib: '1.2.8' }

> name: 'homebridge',
  version: '0.4.16',

> $ dpkg -l |grep cec
ii  cec-utils ____ 3.0.1-1\~jessie _____ armhf _____ USB CEC Adaptor communication Library (utility programs)
ii  libcec3:armhf _____ 3.0.1-1\~jessie ____ armhf _____ USB CEC Adaptor communication Library (shared library)

Tell my what else do you need

**Comment by jbree** [comment]
Great, thanks for the detailed info.Two more requests:

1) can you try running `cec-client` manually and paste the output. I'm wondering if the problem is with `cec-client`, or `node-cec`'s parsing of the `cec-client` output.

2) would you mind running `npm version` in the directory where you have `homebridge-platform-cec` installed so it lists versions for all its dependencies.

I appreciate your help on this. I'd really like to get this issue figured out.

**Comment by sindudas** [comment]
I just tried, making these modifications:

> /usr/local/lib/node_modules/homebridge-platform-cec/node_modules/cec-promise# diff index.js.orig index.js
14c14
\<   }, 5000);
\---
\>   }, 10000);
19c19
\<     }, 3000);
\---
\>     }, 6000);

Increasing x2 the these timeouts for testing. And the results are:

>Jan 26 11:42:33 homebridge[15670]: Receiver getOn: 0
Jan 26 11:42:33 homebridge[15670]: [1/26/2017, 11:42:33 AM] [Led Cortinas] power is currently OFF
Jan 26 11:42:33 homebridge[15670]: [1/26/2017, 11:42:33 AM] [Led Espejo] ... hue is currently 0
Jan 26 11:42:33 homebridge[15670]: [1/26/2017, 11:42:33 AM] [Led Cortinas] ... hue is currently 0
Jan 26 11:42:34 homebridge[15670]: Error: No REPORT_AUDIO_STATUS received after 1000ms
Jan 26 11:42:34 homebridge[15670]: at Timeout._onTimeout (/usr/local/lib/node_modules/homebridge-platform-cec/node_modules/cec-promise/index.js:62:23)
Jan 26 11:42:34 homebridge[15670]: at ontimeout (timers.js:365:14)
Jan 26 11:42:34 homebridge[15670]: at tryOnTimeout (timers.js:237:5)
Jan 26 11:42:34 homebridge[15670]: at Timer.listOnTimeout (timers.js:207:5)

And when trying to power on the TV:
>Jan 26 11:42:55 homebridge[15670]: Receiver getOn: 0
Jan 26 11:42:55 homebridge[15670]: Receiver setOn: 1

But I think it doesn't power on. I'm not at home right now, but the power consumption monitor (emoncms) doesn't report any increase in energy consumption power. And when the homekit refresh the state of the TV reports it's Off:
>Jan 26 11:43:25 homebridge[15670]: Receiver getOn: 0

**Comment by sindudas** [comment]
In Reply to:
> 1. can you try running cec-client manually and paste the output. I'm wondering if the problem is with cec-client, or node-cec's parsing of the cec-client output.

Here the output:
>$ cec-client
No device type given. Using 'recording device'
CEC Parser created - libCEC version 3.0.1
no serial port given. trying autodetect: 
 path:     Raspberry Pi
 com port: RPI
>
>opening a connection to the CEC adapter...
DEBUG:   [             103]	Broadcast (F): osd name set to 'Broadcast'
DEBUG:   [             105]	InitHostCEC - vchiq_initialise succeeded
DEBUG:   [             107]	InitHostCEC - vchi_initialise succeeded
DEBUG:   [             109]	InitHostCEC - vchi_connect succeeded
DEBUG:   [             113]	logical address changed to Free use (e)
DEBUG:   [             115]	Open - vc_cec initialised
NOTICE:  [             116]	connection opened
DEBUG:   [             118]	<< Broadcast (F) -> TV (0): POLL
DEBUG:   [             118]	initiator 'Broadcast' is not supported by the CEC adapter. using 'Free use' instead
TRAFFIC: [             118]	<< e0
DEBUG:   [             122]	processor thread started
DEBUG:   [             149]	>> POLL sent
DEBUG:   [             149]	TV (0): device status changed into 'present'
DEBUG:   [             149]	<< requesting vendor ID of 'TV' (0)
TRAFFIC: [             149]	<< e0:8c
TRAFFIC: [             356]	>> 0f:87:00:e0:91
DEBUG:   [             356]	TV (0): vendor = LG (00e091)
DEBUG:   [             357]	expected response received (87: device vendor id)
DEBUG:   [             357]	replacing the command handler for device 'TV' (0)
DEBUG:   [             357]	TV (0): CEC version 1.3a
DEBUG:   [             357]	TV (0): menu language set to 'eng'
NOTICE:  [             358]	registering new CEC client - v3.0.1
DEBUG:   [             358]	detecting logical address for type 'recording device'
DEBUG:   [             358]	trying logical address 'Recorder 1'
DEBUG:   [             358]	<< Recorder 1 (1) -> Recorder 1 (1): POLL
TRAFFIC: [             358]	<< 11
DEBUG:   [             361]	>> TV (0) -> Broadcast (F): device vendor id (87)
TRAFFIC: [             629]	<< 11
DEBUG:   [             899]	>> POLL not sent
DEBUG:   [             899]	using logical address 'Recorder 1'
DEBUG:   [             899]	Recorder 1 (1): device status changed into 'handled by libCEC'
DEBUG:   [             899]	Recorder 1 (1): power status changed from 'unknown' to 'on'
DEBUG:   [             899]	Recorder 1 (1): vendor = Pulse Eight (001582)
DEBUG:   [             899]	Recorder 1 (1): CEC version 1.4
DEBUG:   [             900]	AllocateLogicalAddresses - device '0', type 'recording device', LA '1'
DEBUG:   [             901]	logical address changed to Recorder 1 (1)
DEBUG:   [             901]	Recorder 1 (1): osd name set to 'CECTester'
DEBUG:   [             901]	Recorder 1 (1): menu language set to 'eng'
DEBUG:   [             902]	GetPhysicalAddress - physical address = 1000
DEBUG:   [             902]	AutodetectPhysicalAddress - autodetected physical address '1000'
DEBUG:   [             902]	Recorder 1 (1): physical address changed from ffff to 1000
DEBUG:   [             902]	<< Recorder 1 (1) -> broadcast (F): physical adddress 1000
TRAFFIC: [             902]	<< 1f:84:10:00:01
NOTICE:  [            1053]	CEC client registered: libCEC version = 3.0.1, client version = 3.0.1, firmware version = 1, logical address(es) = Recorder 1 (1) , physical address: 1.0.0.0,  compiled on Fri Jul  3 19:20:43 UTC 2015 by pbuilder@mgorven-pc.thefacebook.com on Linux 3.16.0-41-generic (armv7l), features: P8_USB, P8_detect, 'RPi'
DEBUG:   [            1053]	Recorder 1 (1): vendor = LG (00e091)
DEBUG:   [            1053]	replacing the command handler for device 'Recorder 1' (1)
DEBUG:   [            1053]	Recorder 1 (1): CEC version 1.3a
DEBUG:   [            1054]	<< Recorder 1 (1) -> TV (0): OSD name 'CECTester'
TRAFFIC: [            1054]	<< 10:47:43:45:43:54:65:73:74:65:72
DEBUG:   [            1355]	<< requesting power status of 'TV' (0)
TRAFFIC: [            1355]	<< 10:8f
waiting for input
TRAFFIC: [            1960]	>> 01:90:00
DEBUG:   [            1960]	TV (0): power status changed from 'unknown' to 'on'
DEBUG:   [            1963]	expected response received (90: report power status)
DEBUG:   [            1966]	>> TV (0) -> Recorder 1 (1): report power status (90)
TRAFFIC: [            9499]	>> 0f:87:00:e0:91
DEBUG:   [            9499]	<< Recorder 1 (1) -> Broadcast (F): vendor id LG (e091)
TRAFFIC: [            9499]	<< 1f:87:00:e0:91
DEBUG:   [            9502]	>> TV (0) -> Broadcast (F): device vendor id (87)
TRAFFIC: [           19499]	>> 0f:87:00:e0:91
DEBUG:   [           19499]	<< Recorder 1 (1) -> Broadcast (F): vendor id LG (e091)
TRAFFIC: [           19499]	<< 1f:87:00:e0:91
DEBUG:   [           19502]	>> TV (0) -> Broadcast (F): device vendor id (87)
TRAFFIC: [           29496]	>> 0f:87:00:e0:91
DEBUG:   [           29496]	<< Recorder 1 (1) -> Broadcast (F): vendor id LG (e091)
TRAFFIC: [           29496]	<< 1f:87:00:e0:91
DEBUG:   [           29500]	>> TV (0) -> Broadcast (F): device vendor id (87)


In reply to:
>would you mind running npm version in the directory where you have homebridge-platform-cec installed so it lists versions for all its dependencies.

I'm not sure if this is what you wanted me to do:
>/usr/local/lib/node_modules/homebridge-platform-cec# npm version
{ 'homebridge-platform-cec': '0.0.1',
  npm: '3.10.10',
  ares: '1.10.1-DEV',
  http_parser: '2.7.0',
  icu: '57.1',
  modules: '48',
  node: '6.9.4',
  openssl: '1.0.2j',
  uv: '1.9.1',
  v8: '5.1.281.89',
  zlib: '1.2.8' }

**Comment by jbree** [comment]
EDIT:
Oh my, I completely missed your first comment when I wrote this, but it looks like you and I came to the same conclusion: that the timeout was the issue.

Here's my original post, bask in my lack of situational awareness:

> Okay, I think this has a stupid-simple solution that I was looking right past. I wrapped the [node-cec](https://github.com/patlux/node-cec) module to use promises in [cec-promise](https://github.com/jbree/cec-promise), but added a 5 second timeout to bail out if the client never reported ready.
> 
> My best guess is that 5 seconds just isn't quite enough for your rpi to get it done.
> 
> I'll be able to adjust timeout tonight and publish, or you can adjust it manually before then to see if it fixes your issue. Here is the [offending line of code](https://github.com/jbree/cec-promise/blob/master/index.js#L14). Also, I see that your goal is to turn on/off your TV. Right now I only have the module built to control the receiver, but I should be able to get the TV power control working in short order.

**Comment by jbree** [comment]
Opened a new issue ( #5 ) for TV support. Will be easy enough to add.

It's probably a good idea for the plugin to load a device list (tv, dvr, receiver, etc) from homebridge config file, rather than assuming your setup. I think that will take care of the REPORT_AUDIO_STATUS timeout error.

**Comment by jbree** [comment]
I've updated cec-promise with that longer timeout, `npm update` should cure that one.

I've added basic support on the [feature-tv-power branch](https://github.com/jbree/homebridge-platform-cec/tree/feature-tv-power), but I want to figure out how to select devices before I release it, right now it's sloppy.

**Comment by sindudas** [comment]
I tried to test the new branch, but I have not yet managed to install it.
If I can help in any way to develop or test, let me know. I'm interested in it working. To control the TV, for the Home Theater, and any else.
Thanks for the work.

**Comment by jbree** [comment]
Thanks for the support! I'm working on it, I'll let you know when I have an update. I'm working out how to specify devices in the homebridge config.json file, and once that's in place we can work out any remaining kinks in your setup.

**Comment by jbree** [comment]
@sindudas, sorry for the extended delay. I decided to cut the "platform" part out of this plugin and just make a manually configurable accessory. Please see my new [homebridge-cec-accessory](https://github.com/jbree/homebridge-cec-accessory) plugin. It will allow you to add controls for only the devices you want, and doesn't require a rewrite to add other devices. It simplifies the code, and makes configuration less of a mystery.

I would greatly appreciate it if you take it for a spin and let me know if it works for you, or if you have any questions about configuration.

**Comment by sindudas** [comment]
Hi @jbree , sorry for delay. I was mounting a RPi3 in parallel to the RPi1. But same results. I played with timeout defined internally, but with no much idea what I was changing.

This is the config with the "homebridge-cec-accessory":

{
      "accessory": "cec",
      "name": "TV Power",
      "type": "power",
      "address": 0
}

With that config, I get on HomeKit 3 objects.
-Light object as "Power", it shows as powered on. Turning on/off doesn't power on/off the TV.
-Light object as "Volume" , but it's unreachable
-Switch object as "TV Power", it shows as powered on. Same behavior as "Power".


>Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Loaded plugin: homebridge-better-http-rgb
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Registering accessory 'homebridge-better-http-rgb.HTTP-RGB'
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] ---
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Loaded plugin: homebridge-cec-accessory
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Registering accessory 'homebridge-cec.cec'
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] ---
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Loaded plugin: homebridge-platform-cec
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Registering platform 'homebridge-platform-cec.CecPlatform'
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] ---
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Loaded config.json with 3 accessories and 0 platforms.
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] ---
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Loading 0 platforms...
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Loading 3 accessories...

>Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] [TV Power] Initializing cec accessory...
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] [TV Power] Creating accessory service
Feb 16 21:05:20 raspberrypi homebridge[6271]: Load homebridge-platform-cec.CecPlatform
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] [homebridge-platform-cec.CecPlatform] homebridge-platform-cec v0.0.1, node v6.9.5, homebridge v2.1
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] [homebridge-platform-cec.CecPlatform] configureAccessory
Feb 16 21:05:20 raspberrypi homebridge[6271]: Updating power service to Sony AV Amp
Feb 16 21:05:20 raspberrypi homebridge[6271]: reconfiguring existing ReceiverPower accessory
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] [homebridge-platform-cec.CecPlatform] configureAccessory
Feb 16 21:05:20 raspberrypi homebridge[6271]: Updating volume service to Sony AV Amp
Feb 16 21:05:20 raspberrypi homebridge[6271]: reconfiguring existing ReceiverVolume accessory
Feb 16 21:05:20 raspberrypi homebridge[6271]: Scan this code with your HomeKit App on your iOS device to pair with Homebridge:
Feb 16 21:05:20 raspberrypi homebridge[6271]: ┌────────────┐
Feb 16 21:05:20 raspberrypi homebridge[6271]: └────────────┘
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] [homebridge-platform-cec.CecPlatform] didFinishLaunching
Feb 16 21:05:20 raspberrypi homebridge[6271]: [2/16/2017, 9:05:20 PM] Homebridge is running on port 51826.

Here is when I opened the HomeKit App and it refresh the status.
>Feb 16 21:11:38 raspberrypi homebridge[6271]: [2/16/2017, 9:11:38 PM] [TV Power] getOn: 224
Feb 16 21:11:38 raspberrypi homebridge[6271]: Receiver getOn: 0
Feb 16 21:11:38 raspberrypi homebridge[6271]: [2/16/2017, 9:11:38 PM] [TV Power] TV Power getOn: 0
Feb 16 21:11:51 raspberrypi homebridge[6271]: Error: No REPORT_AUDIO_STATUS received after 5000ms
Feb 16 21:11:51 raspberrypi homebridge[6271]: at Timeout._onTimeout (/usr/lib/node_modules/homebridge-platform-cec/node_modules/cec-promise/index.js:63:23)
Feb 16 21:11:51 raspberrypi homebridge[6271]: at ontimeout (timers.js:365:14)
Feb 16 21:11:51 raspberrypi homebridge[6271]: at tryOnTimeout (timers.js:237:5)
Feb 16 21:11:51 raspberrypi homebridge[6271]: at Timer.listOnTimeout (timers.js:207:5)

Here I turn on the Light "Power"
>Feb 16 21:14:47 raspberrypi homebridge[6271]: Receiver getOn: 0
Feb 16 21:14:47 raspberrypi homebridge[6271]: Receiver setOn: 1

Here I tried to turn off the Light "Power"
>Feb 16 21:14:59 raspberrypi homebridge[6271]: Error: No REPORT_POWER_STATUS received after 5000ms
Feb 16 21:14:59 raspberrypi homebridge[6271]: at Timeout._onTimeout (/usr/lib/node_modules/homebridge-platform-cec/node_modules/cec-promise/index.js:63:23)
Feb 16 21:14:59 raspberrypi homebridge[6271]: at ontimeout (timers.js:365:14)
Feb 16 21:14:59 raspberrypi homebridge[6271]: at tryOnTimeout (timers.js:237:5)
Feb 16 21:14:59 raspberrypi homebridge[6271]: at Timer.listOnTimeout (timers.js:207:5)
Feb 16 21:14:59 raspberrypi homebridge[6271]: Error: No REPORT_POWER_STATUS received after 5000ms
Feb 16 21:14:59 raspberrypi homebridge[6271]: at Timeout._onTimeout (/usr/lib/node_modules/homebridge-platform-cec/node_modules/cec-promise/index.js:63:23)
Feb 16 21:14:59 raspberrypi homebridge[6271]: at ontimeout (timers.js:365:14)
Feb 16 21:14:59 raspberrypi homebridge[6271]: at tryOnTimeout (timers.js:237:5)
Feb 16 21:14:59 raspberrypi homebridge[6271]: at Timer.listOnTimeout (timers.js:207:5)

Here I tried to turn on the Switch "TV Power"
>Feb 16 21:19:26 raspberrypi homebridge[6271]: [2/16/2017, 9:19:26 PM] [TV Power] getOn: 224
Feb 16 21:19:31 raspberrypi homebridge[6271]: [2/16/2017, 9:19:31 PM] [TV Power] Error: No REPORT_POWER_STATUS received after 5000ms
Feb 16 21:19:31 raspberrypi homebridge[6271]: at Timeout._onTimeout (/usr/lib/node_modules/homebridge-cec-accessory/node_modules/cec-promise/index.js:63:23)
Feb 16 21:19:31 raspberrypi homebridge[6271]: at ontimeout (timers.js:365:14)
Feb 16 21:19:31 raspberrypi homebridge[6271]: at tryOnTimeout (timers.js:237:5)
Feb 16 21:19:31 raspberrypi homebridge[6271]: at Timer.listOnTimeout (timers.js:207:5)
Feb 16 21:19:31 raspberrypi homebridge[6271]: [2/16/2017, 9:19:31 PM] [TV Power] Error: No REPORT_POWER_STATUS received after 5000ms
Feb 16 21:19:31 raspberrypi homebridge[6271]: at Timeout._onTimeout (/usr/lib/node_modules/homebridge-cec-accessory/node_modules/cec-promise/index.js:63:23)
Feb 16 21:19:31 raspberrypi homebridge[6271]: at ontimeout (timers.js:365:14)
Feb 16 21:19:31 raspberrypi homebridge[6271]: at tryOnTimeout (timers.js:237:5)
Feb 16 21:19:31 raspberrypi homebridge[6271]: at Timer.listOnTimeout (timers.js:207:5)


Thanks for working on it

**Comment by jbree** [comment]
@sindudas,
It looks like you still have the homebridge-platform-cec plugin installed, and the platform configuration in your `config.json` file. I think I was unclear about my intent with homebridge-cec-accessory. It is a replacement for, not an augmentation of, homebridge-platform-cec. Please delete the configuration in the `platforms` section of your `config.json` and update me with your results.

**Comment by sindudas** [comment]
No, platform config was empty:

> "platforms": []

**Comment by sindudas** [comment]
Now, uninstalled homebridge-platform-cec

>Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] Loaded plugin: homebridge-better-http-rgb
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] Registering accessory 'homebridge-better-http-rgb.HTTP-RGB'
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] ---
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] Loaded plugin: homebridge-cec-accessory
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] Registering accessory 'homebridge-cec.cec'
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] ---
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] Loaded config.json with 3 accessories and 0 platforms.
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] ---
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] Loading 0 platforms...
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] Loading 3 accessories...
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] [TV Power] Initializing cec accessory...
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] [TV Power] Creating accessory service
Feb 16 21:35:39 raspberrypi homebridge[7003]: Scan this code with your HomeKit App on your iOS device to pair with Homebridge:
Feb 16 21:35:39 raspberrypi homebridge[7003]: ┌────────────┐
Feb 16 21:35:39 raspberrypi homebridge[7003]: └────────────┘
Feb 16 21:35:39 raspberrypi homebridge[7003]: [2/16/2017, 9:35:39 PM] Homebridge is running on port 51826.


Erased accessories/cachedAccessories to disappear it:
>Feb 16 21:35:39 raspberrypi homebridge[7003]: Failed to find plugin to handle accessory Sony AV Amp
Feb 16 21:35:39 raspberrypi homebridge[7003]: Failed to find plugin to handle accessory Sony AV Amp

**Comment by jbree** [comment]
Ah, okay! Homebridge caches previously used accessories and configuration. Best bet would be to nuke those configurations, delete the Pi from your "Home" app, and run through setup again. In case you're unfamiliar, you delete the cached stuff by deleting `.homebridge/persist` and `.homebridge/accessories`.

**Comment by jbree** [comment]
Then run it again, and set it up on the Home app again

**Comment by sindudas** [comment]
Sorry. All cleaned and adding homebridge to Home app again

>Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:10 PM] Loaded plugin: homebridge-better-http-rgb
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] Registering accessory 'homebridge-better-http-rgb.HTTP-RGB'
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] ---
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] Loaded plugin: homebridge-cec-accessory
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] Registering accessory 'homebridge-cec.cec'
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] ---
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] Loaded config.json with 3 accessories and 0 platforms.
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] ---
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] Loading 0 platforms...
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] Loading 3 accessories...
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] [TV Power] Initializing cec accessory...
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] [TV Power] Creating accessory service
Feb 16 21:42:11 raspberrypi homebridge[7283]: Scan this code with your HomeKit App on your iOS device to pair with Homebridge:
Feb 16 21:42:11 raspberrypi homebridge[7283]: ┌────────────┐
Feb 16 21:42:11 raspberrypi homebridge[7283]: └────────────┘
Feb 16 21:42:11 raspberrypi homebridge[7283]: [2/16/2017, 9:42:11 PM] Homebridge is running on port 51826.
Feb 16 21:42:29 raspberrypi homebridge[7283]: [2/16/2017, 9:42:29 PM] [TV Power] getOn: 224
Feb 16 21:42:34 raspberrypi homebridge[7283]: [2/16/2017, 9:42:34 PM] [TV Power] Error: No REPORT_POWER_STATUS received after 5000ms
Feb 16 21:42:34 raspberrypi homebridge[7283]: at Timeout._onTimeout (/usr/lib/node_modules/homebridge-cec-accessory/node_modules/cec-promise/index.js:63:23)
Feb 16 21:42:34 raspberrypi homebridge[7283]: at ontimeout (timers.js:365:14)
Feb 16 21:42:34 raspberrypi homebridge[7283]: at tryOnTimeout (timers.js:237:5)
Feb 16 21:42:34 raspberrypi homebridge[7283]: at Timer.listOnTimeout (timers.js:207:5)


"TV Power" Switch shows "unreachable"

**Comment by jbree** [comment]
Are you loading any other accessories that are using CEC? It looks like you might be loading `homebridge-cec` as well. the RPI can only have one instance of the `cec-client` application running at a time, and `homebridge-cec` and `homebridge-cec-accessory` both try to use it.

Another troubleshooting step you can do is to try shutting down homebridge and run `cec-client`. Once it reports `waiting for input`, type `tx e0:8f` and press return to request the power status from your TV.

This is a good reference for which commands you can send, and how they are addressed: http://www.cec-o-matic.com/

**Comment by sindudas** [comment]
Now I don't have installed the "homebridge-cec" only "homebrige-cec-accessory".
And no other cec accessories. Only two Led accessories.

Only one cdc-client running now:
>homebri+  7950  0.1  0.3  66500  3248 ?        Sl   22:07   0:00 cec-client -m -d 8 -b r -o node-cec-monitor

Stop homebridge and run "cec-client" and type that:
>TRAFFIC: [           14805]	<< e0:8f
WARNING: [           14955]	unhandled response received: opcode=8f initiator=1 destination=0 response=0
TRAFFIC: [           15037]	>> 01:90:00
DEBUG:   [           15037]	>> TV (0) -> Recorder 1 (1): report power status (90)
DEBUG:   [           15805]	command 'give device power status' timeout
waiting for input

**Comment by jbree** [comment]
I'm really happy you're sticking around to help me work through these issues.

I think the sender address is wrong, and your TV is refusing to respond to the message because of that fact. Run `cec-client` again, and this time do `tx 10:8f`. If this reports the status back correctly, I have a code modification to potentially fix it:

Navigate to homebridge-cec-accessory in your global `node_modules` folder, and modify `dist/controls/Control.js` to show `this.senderNibble = 1;` rather than `14`. According to your logs above, 1 is the correct address for your rpi.

The library I'm using doesn't provide a way to detect it this address, and none of my personal devices care about the mismatch between reported and actual address, so I just left it at 14. If this fix pans out for you, I'll look into a method for detecting the rpi cec address if possible, else manually specifying.

**Comment by sindudas** [comment]
Now arrived to home, Home App shows "TV Power" as switched On. Turning it "off", shows in log:
>Feb 17 19:28:00 raspberrypi homebridge[751]: [2/17/2017, 7:28:00 PM] [TV Power] getOn: 224
Feb 17 19:28:01 raspberrypi homebridge[751]: [2/17/2017, 7:28:01 PM] [TV Power] TV Power getOn: 1
Feb 17 19:28:01 raspberrypi homebridge[751]: [2/17/2017, 7:28:01 PM] [TV Power] TV Power setOn: 0

Turning it "on", shows this:
>Feb 17 19:28:08 raspberrypi homebridge[751]: [2/17/2017, 7:28:08 PM] [TV Power] getOn: 224
Feb 17 19:28:09 raspberrypi homebridge[751]: [2/17/2017, 7:28:09 PM] [TV Power] TV Power getOn: 1
Feb 17 19:28:09 raspberrypi homebridge[751]: [2/17/2017, 7:28:09 PM] [TV Power] TV Power setOn: already on


Now, the TV is in Standby.

Power On the TV, and the Log shows as this:

>Feb 17 19:31:51 raspberrypi homebridge[751]: [2/17/2017, 7:31:51 PM] [TV Power] Error: No REPORT_POWER_STATUS received after 10000ms
Feb 17 19:31:51 raspberrypi homebridge[751]: at Timeout._onTimeout (/usr/lib/node_modules/homebridge-cec-accessory/node_modules/cec-promise/index.js:63:23)
Feb 17 19:31:51 raspberrypi homebridge[751]: at ontimeout (timers.js:365:14)
Feb 17 19:31:51 raspberrypi homebridge[751]: at tryOnTimeout (timers.js:237:5)
Feb 17 19:31:51 raspberrypi homebridge[751]: at Timer.listOnTimeout (timers.js:207:5)


Now I'm going to do last suggestions to do.

**Comment by sindudas** [comment]
One step done. My TV doesn't like to ask "e0:8f" when it is on Standby. But when it is powered on, it replies to "e0:8f". Asking "10:8f", with TV on or off, always gets the response:


>$ echo "10:8f" | cec-client -s
opening a connection to the CEC adapter...
DEBUG:   [              55]	Broadcast (F): osd name set to 'Broadcast'
DEBUG:   [              55]	InitHostCEC - vchiq_initialise succeeded
DEBUG:   [              55]	InitHostCEC - vchi_initialise succeeded
DEBUG:   [              55]	InitHostCEC - vchi_connect succeeded
DEBUG:   [              56]	Open - vc_cec initialised
DEBUG:   [              56]	logical address changed to Free use (e)
NOTICE:  [              56]	connection opened
DEBUG:   [              57]	processor thread started
DEBUG:   [              57]	<< Broadcast (F) -> TV (0): POLL
DEBUG:   [              57]	initiator 'Broadcast' is not supported by the CEC adapter. using 'Free use' instead
TRAFFIC: [              57]	<< e0
DEBUG:   [             117]	>> POLL sent
DEBUG:   [             117]	TV (0): device status changed into 'present'
DEBUG:   [             117]	<< requesting vendor ID of 'TV' (0)
TRAFFIC: [             117]	<< e0:8c
TRAFFIC: [             308]	>> 0f:87:00:e0:91
DEBUG:   [             309]	>> TV (0) -> Broadcast (F): device vendor id (87)
DEBUG:   [             309]	TV (0): vendor = LG (00e091)
DEBUG:   [             309]	expected response received (87: device vendor id)
DEBUG:   [             309]	replacing the command handler for device 'TV' (0)
DEBUG:   [             309]	TV (0): CEC version 1.3a
DEBUG:   [             309]	TV (0): menu language set to 'eng'
NOTICE:  [             309]	registering new CEC client - v3.0.1
DEBUG:   [             309]	detecting logical address for type 'recording device'
DEBUG:   [             309]	trying logical address 'Recorder 1'
DEBUG:   [             309]	<< Recorder 1 (1) -> Recorder 1 (1): POLL
TRAFFIC: [             309]	<< 11
TRAFFIC: [             580]	<< 11
DEBUG:   [             850]	>> POLL not sent
DEBUG:   [             850]	using logical address 'Recorder 1'
DEBUG:   [             850]	Recorder 1 (1): device status changed into 'handled by libCEC'
DEBUG:   [             850]	Recorder 1 (1): power status changed from 'unknown' to 'on'
DEBUG:   [             850]	Recorder 1 (1): vendor = Pulse Eight (001582)
DEBUG:   [             850]	Recorder 1 (1): CEC version 1.4
DEBUG:   [             850]	AllocateLogicalAddresses - device '0', type 'recording device', LA '1'
DEBUG:   [             851]	logical address changed to Recorder 1 (1)
DEBUG:   [             851]	Recorder 1 (1): osd name set to 'CECTester'
DEBUG:   [             851]	Recorder 1 (1): menu language set to 'eng'
DEBUG:   [             851]	GetPhysicalAddress - physical address = 1000
DEBUG:   [             851]	AutodetectPhysicalAddress - autodetected physical address '1000'
DEBUG:   [             851]	Recorder 1 (1): physical address changed from ffff to 1000
DEBUG:   [             851]	<< Recorder 1 (1) -> broadcast (F): physical adddress 1000
TRAFFIC: [             852]	<< 1f:84:10:00:01
NOTICE:  [            1002]	CEC client registered: libCEC version = 3.0.1, client version = 3.0.1, firmware version = 1, logical address(es) = Recorder 1 (1) , physical address: 1.0.0.0,  compiled on Fri Jul  3 19:20:43 UTC 2015 by pbuilder@mgorven-pc.thefacebook.com on Linux 3.16.0-41-generic (armv7l), features: P8_USB, P8_detect, 'RPi'
DEBUG:   [            1002]	Recorder 1 (1): vendor = LG (00e091)
DEBUG:   [            1002]	replacing the command handler for device 'Recorder 1' (1)
DEBUG:   [            1002]	Recorder 1 (1): CEC version 1.3a
DEBUG:   [            1002]	<< Recorder 1 (1) -> TV (0): OSD name 'CECTester'
TRAFFIC: [            1002]	<< 10:47:43:45:43:54:65:73:74:65:72
DEBUG:   [            1303]	<< requesting power status of 'TV' (0)
TRAFFIC: [            1303]	<< 10:8f
TRAFFIC: [            1445]	>> 01:90:00
DEBUG:   [            1446]	>> TV (0) -> Recorder 1 (1): report power status (90)
DEBUG:   [            1446]	TV (0): power status changed from 'unknown' to 'on'
DEBUG:   [            1446]	expected response received (90: report power status)
DEBUG:   [            1447]	unregistering all CEC clients
NOTICE:  [            1447]	unregistering client: libCEC version = 3.0.1, client version = 3.0.1, firmware version = 1, logical address(es) = Recorder 1 (1) , physical address: 1.0.0.0,  compiled on Fri Jul  3 19:20:43 UTC 2015 by pbuilder@mgorven-pc.thefacebook.com on Linux 3.16.0-41-generic (armv7l), features: P8_USB, P8_detect, 'RPi'
DEBUG:   [            1447]	Recorder 1 (1): power status changed from 'on' to 'unknown'
DEBUG:   [            1447]	Recorder 1 (1): vendor = Unknown (000000)
DEBUG:   [            1447]	Recorder 1 (1): CEC version unknown
DEBUG:   [            1447]	Recorder 1 (1): osd name set to 'Recorder 1'
DEBUG:   [            1447]	Recorder 1 (1): device status changed into 'unknown'
DEBUG:   [            1447]	unregistering all CEC clients
DEBUG:   [            2057]	UnregisterLogicalAddress - releasing previous logical address
DEBUG:   [            2058]	logical address changed to Broadcast (f)

**Comment by jbree** [comment]
Excellent! This means that changing that line in `Control.js` to `this.senderNibble = 1;` should work. I'll put in some time this weekend to find a permanent, flexible solution.

**Comment by sindudas** [comment]
The problem I found, is that LG TVs (what I got) doesn't do the action "standby". The HomeTheater I think it will do.

Here is the compatibility table I found about that problem
[http://libcec.pulse-eight.com/vendor/support](http://libcec.pulse-eight.com/vendor/support)

**Comment by sindudas** [comment]
Seems that LG TVs can't put in standby, all models. To not try this unsupported feature, you can identify with the vendors ID and ignore that command or else.
I have to find another way to power it off from homebridge :P 

Next step will be test the HomeTheather, maybe I will be more lucky with it. I will report my results.

**Comment by jbree** [comment]
That is too bad :( I hope your audio system will work, at least.

Before I started with these cec projects, I was using an IR LED and an ESP8266 microcontroller to send control codes to my devices. I became interested in CEC because I wanted device status *and* control, and the LED only gave me control. Unfortunately, it looks like you have the opposite problem: status, but only partial control.

**Comment by sindudas** [comment]
If you can share any link wik ESP to control the Power Off with IR, I will appreciate. If I do the Power Off that way, and with CEC Status and Power On, I will be "full" control on the TV and the others.

**Comment by jbree** [comment]
Since you have a raspberry pi already running homebridge, I think the [LIRC](http://lirc.org) project might be a good starting reference point. Running something there on the pi would probably be easier than setting up a server/client with esp8266. Some other folks have done some nodejs and homebridge projects with it, which I'm sure could be adapted to your needs.
